src/components/TablePagination.tsx

```diff
@@ -0,0 +1,31 @@
+import { Table } from '@tanstack/react-table';
+import { Button } from './ui/button';
+
+type TablePaginationProps<TData> = {
+  table: Table<TData>;
+};
+
+export const TablePagination = <TData,>({
+  table,
+}: TablePaginationProps<TData>) => {
+  return (
+    <div className="flex items-center justify-end space-x-2 py-4">
+      <Button
+        variant="outline"
+        size="sm"
+        onClick={() => table.previousPage()}
+        disabled={!table.getCanPreviousPage()}
+      >
+        Předchozí
+      </Button>
+      <Button
+        variant="outline"
+        size="sm"
+        onClick={() => table.nextPage()}
+        disabled={!table.getCanNextPage()}
+      >
+        Další
+      </Button>
+    </div>
```

Comment:
Sending the `table` object as a prop is not a good idea. This component doesn't need the complete object just some small parts of it. Imagine if you would decide to replace TanstackTable with a different solution, then you would need to change this component also.

It's better to define handler functions in the parent component and pass them as props. This will decuple the pagination component from implementation details of the table provider.
```ts
 const getPreviousPage = () => dataTable.previousPage();
  const getNextPage = () => dataTable.nextPage();
  const isPreviousPageDisabled = !dataTable.getCanPreviousPage();
  const isNextPageDisabled = !dataTable.getCanNextPage();
```